src/spikeinterface/curation/train_manual_curation.py

```diff
@@ -0,0 +1,375 @@
+        )
+
+        model_name = "noise" if objective == Objective.Noise else "sua"
+
+        # Save the pipeline with pickle
+        pkl.dump(best_pipeline, open(os.path.join(self.output_folder, setting, f"best_model_{model_name}.pkl"), "wb"))
+
+        test_accuracies_df.to_csv(
+            os.path.join(self.output_folder, setting, f"model_{model_name}_accuracies.csv"), float_format="%.4f"
+        )
+
+    def _train_and_evaluate(
+        self, imputation_strategy, scaler_name, scaler, classifier, X_train, X_test, y_train, y_test, model_id
+    ):
+        from skopt import BayesSearchCV
+        from sklearn.metrics import balanced_accuracy_score, precision_score, recall_score
+
+        X_train_scaled, X_test_scaled, y_train, y_test, imputer, scaler = self.apply_scaling_imputation(
+            imputation_strategy, scaler, X_train, X_test, y_train, y_test
+        )
+        logger.debug(f"Running {classifier} with imputation {imputation_strategy} and scaling {scaler_name}")
+        model, param_space = self.get_classifier_search_space(classifier)
+        model = BayesSearchCV(
+            model, param_space, cv=3, scoring="balanced_accuracy", n_iter=25, random_state=seed, n_jobs=-1
+        )
+        model.fit(X_train_scaled, y_train)
+        y_pred = model.predict(X_test_scaled)
+        balanced_acc = balanced_accuracy_score(y_test, y_pred)
+        precision = precision_score(y_test, y_pred, average="macro")
+        recall = recall_score(y_test, y_pred, average="macro")
+        return {
+            "classifier name": classifier.__name__,
+            "imputation_strategy": imputation_strategy,
+            "scaling_strategy": scaler_name,
+            "accuracy": balanced_acc,
+            "precision": precision,
+            "recall": recall,
+            "model_id": model_id,
+            "best_params": model.best_params_,
+        }, (model, imputer, scaler)
+
+
+def train_model(metrics_path, output_folder, objective_label):
+
+    from sklearn.preprocessing import StandardScaler, MinMaxScaler, RobustScaler
+
+    if objective_label == "is_noise":
+        objective = Objective.Noise
+    elif objective_label == "is_sua":
+        objective = Objective.SUA
+    else:
+        raise ValueError("Unknown objective label")
+
+    trainer = CurationModelTrainer(objective_label, output_folder)
+    trainer.load_and_preprocess_full(metrics_path)
+
+    imputation_strategies = ["median", "most_frequent", "knn", "iterative"]
+    scaling_techniques = [
+        ("standard_scaler", StandardScaler()),
+        ("min_max_scaler", MinMaxScaler()),
+        ("robust_scaler", RobustScaler()),
+    ]
+
+    classifiers = [
+        RandomForestClassifier,
+        AdaBoostClassifier,
+        GradientBoostingClassifier,
+        SVC,
+        LogisticRegression,
+        XGBClassifier,
+        CatBoostClassifier,
+        LGBMClassifier,
+        MLPClassifier,
+    ]
+
+    metrics_list = [
```

Comment:
This might be better passed to the function by the user as an optional argument?
```suggestion
    def train_model(metrics_path, output_folder, objective_label, metrics_list=None):
        metrics_list = metrics_list if metrics_list is not None else ["num_spikes", ...
```